### ISSUE TITLE
Allow access to field value in Layer.enabled()

### ISSUE THREAD

**Issue by sjoshid** [tokio-rs/tracing#1293]
## Feature Request
API change of `Layer.enabled()` to include `Attributes`

### Crates
[Tracing subscriber](https://docs.rs/crate/tracing-subscriber/0.2.16)

### Motivation
I have a custom layer that wants to filter out a span based on a value of a field. `register_callsite()` of the layer always returns `Interest::sometimes()` and in `enabled()` I want to return true/false based on a field value.

### Proposal
NA

### Alternatives
None. (That I can think of)

**Comment by hawkw** [comment]
The `enabled` method is called _before_ actually recording field values. This is for performance reasons — `enabled`-based filtering is intended to disable a span or event as early as possible, before we do most of the work of actually recording that trace; if field values are expensive to calculate, calling `enabled` before recording the values lets us skip them entirely.

If you want to filter out what spans are actually displayed or recorded by the subscriber, after field values are recorded, the best way would be to add some code to your layer's [`new_span` method](https://docs.rs/tracing-subscriber/0.2.16/tracing_subscriber/layer/trait.Layer.html#method.new_span) that determines if a span contains an unwanted value, and adds a [span extension](https://docs.rs/tracing-subscriber/0.2.16/tracing_subscriber/registry/trait.SpanData.html#tymethod.extensions_mut) indicating that the span is disabled. In any other methods your layer implements, you could test for that extension's presence, and return early if it's set.

In `tracing-core` 0.2, we're considering changing the `new_span` method to return an `Option<Id>` rather than an `Id`, allowing the subscriber an additional opportunity to disable spans based on field values. That would neatly solve this use-case, but it won't be possible until 0.2 is ready. See #924 for details on that.

**Comment by sjoshid** [comment]
> In any other methods your layer implements, you could test for that extension's presence, and return early if it's set.

Returning early in other methods is fine, but wouldnt the subscriber still create a span (albeit infinitesimal)? What Im trying to do is not create a span at all. 
I understand what Im trying to do is not possible. I just want to manage my expectations.

**Comment by hawkw** [comment]
Yeah, that's correct. It's currently not possible to completely disable a span after its fields are recorded. The issue I linked above would make this possible, but it's a breaking change. We'll likely make that change in 0.2.
